src/app/TargetSelect/TargetSelect.tsx

```diff
@@ -63,13 +64,34 @@ export const TargetSelect: React.FunctionComponent<TargetSelectProps> = (props)
   const addSubscription = useSubscriptions();
 
   React.useEffect(() => {
-    const sub = context.targets.targets().subscribe((targets) => {
-      setTargets(targets);
-      selectTargetFromCache(targets);
-    });
-    return () => sub.unsubscribe();
+    addSubscription(
+      context.targets.targets().subscribe((targets) => {
+        setTargets(targets);
+        selectTargetFromCache(targets);
+      })
+    );
   }, [context, context.targets, setTargets]);
 
+  React.useEffect(() => {
+    addSubscription(
+      context.notificationChannel.messages(NotificationCategory.TargetJvmDiscovery)
+        .subscribe(v => {
+          const evt: TargetDiscoveryEvent = v.message.event;
+          if (evt.kind === 'LOST') {
+            const target: Target = {
+              connectUrl: evt.serviceRef.connectUrl,
+              alias: evt.serviceRef.alias,
+            }
+            context.target.target().subscribe((currentTarget) => {
```

Comment:
This subscription is left open. It should either be added to a nested `addSubscription` call, or you can use something like `.pipe(first())` so that it automatically closes after the first value emission.